sdk/core/Azure.Core/src/Shared/NextLinkOperationImplementation.cs

```diff
@@ -274,9 +387,70 @@ private async ValueTask<Response> GetResponseAsync(bool async, string uri, Cance
             {
                 _pipeline.Send(message, cancellationToken);
             }
+
+            // If we are doing final get for a delete LRO with 404, just return empty response with 204
+            if (message.Response.Status == 404 && RequestMethod == RequestMethod.Delete)
+            {
+                return new EmptyResponse(HttpStatusCode.NoContent);
+            }
             return message.Response;
         }
 
+        /// <summary>
+        /// This is only used for final get of the delete LRO, we just want to return an empty response with 204 to the user for this case.
+        /// </summary>
+        private sealed class EmptyResponse : Response
+        {
+            public EmptyResponse(HttpStatusCode status)
+            {
+                Status = (int)status;
+                ReasonPhrase = status.ToString();
+            }
+
+            public override int Status { get; }
+
+            public override string ReasonPhrase { get; }
+
+            public override Stream? ContentStream { get => null; set => throw new System.NotImplementedException(); }
+            public override string ClientRequestId { get => string.Empty; set => throw new System.NotImplementedException(); }
```

Comment:
Would it ever make sense to return a client request id value to help the end-user with tracing?  These are typically used for people investigating ICMs and other issues that require debugging application errors.

Reply:
You are right, it is useful for troubleshooting, added it for user to get.